arangod/RocksDBEngine/RocksDBVPackIndex.cpp

```diff
@@ -1776,19 +1814,25 @@ Result RocksDBVPackIndex::insert(transaction::Methods& trx,
 
     rocksdb::Status s;
     for (RocksDBKey const& key : elements) {
-      if (!isIndexCreation) {
-        // banish key in in-memory cache.
-        // not necessary during index creation, because nothing
-        // will be in the in-memory cache.
-        invalidateCacheEntry(::lookupValueFromSlice(key.string()));
-      }
-
       TRI_ASSERT(key.containsLocalDocumentId(documentId));
       s = mthds->PutUntracked(_cf, key, value.string());
       if (!s.ok()) {
         res = rocksutils::convertStatus(s, rocksutils::index);
         break;
       }
+
+      if (!isIndexCreation) {
+        // banish key in in-memory cache.
+        // not necessary during index creation, because nothing
+        // will be in the in-memory cache.
+        auto slice = ::lookupValueFromSlice(key.string());
+        invalidateCacheEntry(slice);
+        if (_cache != nullptr &&
+            (_forceCacheRefill || options.refillIndexCaches)) {
+          RocksDBTransactionState::toState(&trx)->trackIndexCacheRefill(
+              _collection.id(), id(), {slice.data(), slice.size()});
+        }
```

Comment:
Duplicate of line 1744-1753. Can we move this into a separate function?

Reply:
addressed